src/MongoDB.Driver.Core/Core/Operations/AsyncCursor.cs

```diff
@@ -254,12 +337,9 @@ protected virtual void Dispose(bool disposing)
                 {
                     try
                     {
-                        if (_cursorId != 0)
+                        using (var source = new CancellationTokenSource(TimeSpan.FromSeconds(10)))
```

Comment:
We should encapsulate the 10 second cancellation token and swallowing of exceptions in a CloseIfNotAlreadyCloseFromDispose method.

Reply:
Ok